ivre/xmlnmap.py

```diff
@@ -665,13 +666,75 @@ def change_rpcinfo(table):
             for proto, result in viewitems(protores)]
 
 
+def change_ms_sql_info(table):
+    """Adapt structured output generated by the "ms-sql-info" Nmap script. The
+structured output uses instances (hence, data) as keys, which is undesirable in 
+the databases.
+
+New in SCHEMA_VERSION == 13.
+
+    """
+    instances = []
+    for iname in list(table):
+        idata = table[iname]
+        if not isinstance(idata, dict):
+            continue
+        instances.append(dict(idata, Instance=iname))
+        del table[iname]
+    return instances
+
+
+def change_ms_sql_info_migrate(table):
+    """Adapt structured output generated by the "ms-sql-info" Nmap script. The
+structured output uses instances (hence, data) as keys, which is
+undesirable in the databases.
+
+New in SCHEMA_VERSION == 13.
+
+Use this function when migrating existing records.
+
+In previous schema versions, instances were used has keys; in keys,
+dots are replaced by underscores; this function reverts this change by
+replacing underscores by dots. This is OK because underscores are not
+allowed in FQDNs.
+
+    """
+    instances = []
+    for iname in list(table):
+        idata = table[iname]
+        if not isinstance(idata, dict):
+            continue
+        instances.append(dict(idata, Instance=iname.replace('_', '.')))
+        del table[iname]
+    return instances
+
+
+def change_smb_enum_shares_migrate(table):
```

Comment:
Do you think you could reuse the code from the existing `change_smb_enum_shares()` function? Or is there a reason not to?

Reply:
Yes, I am working on it.